src/js/BookReader/Toolbar/Toolbar.js

```diff
@@ -0,0 +1,383 @@
+    });
+    $form.find('.twitter-share-button').click(() => {
+      const params = $.param({
+        url: this._getSocialShareUrl(),
+        text: br.bookTitle
+      });
+      const url = 'https://twitter.com/intent/tweet?' + params;
+      createPopup(url, 600, 400, 'Share');
+    });
+    $form.find('.email-share-button').click(() => {
+      const body = `${br.bookTitle}\n\n${this._getSocialShareUrl()}`;
+      window.location.href = `mailto:?subject=${encodeURI(br.bookTitle)}&body=${encodeURI(body)}`;
+    });
+
+    $form.find('input[name=thispage]').trigger('change');
+
+    $form.appendTo($shareDiv);
+  }
+
+  _getSocialShareUrl() {
+    const { br } = this;
+    const shareThisPage = br.$('.thispage-social').prop('checked');
+    if (shareThisPage) {
+      return window.location.href;
+    } else {
+      return `${document.location.protocol}//${window.location.hostname}${window.location.pathname}`;
+    }
+  }
+
+  /**
+   * @param {JQuery} $infoDiv DOM element. Appends info to this element
+   * Can be overridden or extended
+   */
+  buildInfoDiv($infoDiv) {
+    const { br } = this;
+    // Remove these legacy elements
+    $infoDiv.find('.BRfloatBody, .BRfloatCover, .BRfloatFoot').remove();
+
+    const $leftCol = $(`<div class="BRinfoLeftCol"></div>`);
+    if (br.thumbnail) {
+      $leftCol.append($(`
+        <div class="BRimageW">
+          <img src="${br.thumbnail}" alt="${escapeHTML(br.bookTitle)}" />
+        </div>`));
+    }
+
+    const $rightCol = $(`<div class="BRinfoRightCol">`);
+
+    // A loop to build fields
+    for (let {extraValueClass='', label, value} of br.metadata) {
+      $rightCol.append($(`
+        <div class="BRinfoValueWrapper">
+          <div class="BRinfoLabel">${label}</div>
+          <div class="BRinfoValue ${extraValueClass}">${value}</div>
+        </div>`));
+    }
+
+    const moreInfoText = br.bookUrlMoreInfo ? br.bookUrlMoreInfo : br.bookTitle;
+    if (moreInfoText && br.bookUrl) {
+      $rightCol.append($(`
+        <div class="BRinfoValueWrapper">
+          <div class="BRinfoMoreInfoWrapper">
+            <a class="BRinfoMoreInfo" href="${br.bookUrl}">
+              ${moreInfoText}
+            </a>
+          </div>
+        </div>`));
+    }
+
+    const $footer = $(`<div class="BRfloatFoot BRinfoFooter"></div>`);
+    const $children = $('<div class="BRinfoW mv20-lg">').append([
+      $leftCol,
+      $rightCol,
+      $('<br style="clear:both"/>'),
+    ]);
+
+    $infoDiv
+      .append($children, $footer)
+      .addClass('wide');
+  }
+
+  /**
+   * @return {number} (in pixels) of the toolbar height. 0 if no toolbar.
+   */
+  getToolBarHeight() {
+    const { $BRtoolbar } = this.br.refs;
+    if ($BRtoolbar && $BRtoolbar.css('display') === 'block') {
+      return ($BRtoolbar.outerHeight() + parseInt($BRtoolbar.css('top')));
+    } else {
+      return 0;
+    }
+  }
+}
+
+export function blankInfoDiv() {
+  // FIXME the <button> el is closed with a </a>
```

Comment:
is this comment still relevant?

Reply:
Yep; I added it during the refactor 😅 I didn't want to figure out what the "correct" thing was (not even sure how the browser is rendering this). Will create an issue :+1:

Reply:
https://github.com/internetarchive/bookreader/issues/240